pkg/cmd/cli/cmd/process.go

```diff
@@ -305,18 +307,23 @@ func RunProcess(f *clientcmd.Factory, in io.Reader, out, errout io.Writer, cmd *
 	}
 	objects = append(objects, resultObj.Objects...)
 
-	p, _, err := kcmdutil.PrinterForCommand(cmd)
+	p, _, err := f.PrinterForCommand(cmd)
 	if err != nil {
 		return err
 	}
-	gv := mapping.GroupVersionKind.GroupVersion()
-	version, err := kcmdutil.OutputVersion(cmd, &gv)
-	if err != nil {
-		return err
+	var version schema.GroupVersion
+	outputVersionString := kcmdutil.GetFlagString(cmd, "output-version")
```

Comment:
No, you must retain this.  Output-version controls how the template is processed and what api group it goes into.  It defaults.